src/OpenSage.Game/Client/Drawable.cs

```diff
@@ -52,12 +36,25 @@ private ModuleBase GetModuleByTag(string tag)
 
         public readonly BitArray<ModelConditionFlag> ModelConditionFlags;
 
-        // Doing this with a field and a property instead of an auto-property allows us to have a read-only public interface,
-        // while simultaneously supporting fast (non-allocating) iteration when accessing the list within the class.
-        public IReadOnlyList<DrawModule> DrawModules => _drawModules;
-        private readonly List<DrawModule> _drawModules;
+        public ReadOnlySpan<DrawModule> DrawModules
+        {
+            get
+            {
+                if (ModelConditionFlags.BitsChanged)
```

Comment:
Previously, we only recomputed model condition states when rendering... which made it hard to set e.g. properties of the animation we've just set. So I've changed it to recompute on-demand, and everything that needs the draw modules must go through this property.